### PR TITLE
feat: Rust

### PR DESCRIPTION
**Pull request by lemaitre-aneo**
API in Rust

### CODE REVIEW

**Review by aneojgurhem — COMMENTED** [review]
remember to setup `cargo fmt`

File: packages/rust/armonik/src/client/agent.rs [hunk 1]
```diff
@@ -0,0 +1,402 @@
...
+
+    pub async fn create_tasks(
+        &mut self,
+        request: impl Stream<Item = create_tasks::Request> + Send + 'static,
+    ) -> Result<Vec<create_tasks::Status>, super::RequestError> {
+        let response = self.call(request).await?;
+
+        match response {
+            create_tasks::Response::Status {
+                communication_token: _,
```
Comment (aneojgurhem) [inline packages/rust/armonik/src/client/agent.rs:101]: I think we need a token here

File: packages/rust/armonik/src/client/sessions.rs [hunk 2]
```diff
@@ -0,0 +1,546 @@
...
+            })
+            .await?
+            .session)
+    }
+
+    /// Close a session by its id.
+    pub async fn close(
+        &mut self,
+        session_id: impl Into<String>,
+    ) -> Result<Raw, super::RequestError> {
```
Comment (aneojgurhem) [inline packages/rust/armonik/src/client/sessions.rs:111]: do we keep Raw or RawSession ?

File: packages/rust/armonik/src/client/events.rs [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+    pub fn with_channel(channel: T) -> Self {
+        Self {
+            inner: v3::events::events_client::EventsClient::new(channel),
+        }
+    }
+
+    /// Get current user
+    pub async fn subscribe(
+        &mut self,
+
```
Comment (aneojgurhem) [inline packages/rust/armonik/src/client/events.rs:32]:
```suggestion
```

File: packages/rust/armonik/examples/client.rs
Comment (aneojgurhem) [inline packages/rust/armonik/examples/client.rs:1]: to me, samples should be in ArmoniK.Samples

File: packages/rust/armonik/src/client/agent.rs (hunk 1 above)
Comment (lemaitre-aneo) [reply packages/rust/armonik/src/client/agent.rs:101]: No, I don't return the token. Here it is the response.

File: packages/rust/armonik/src/client/sessions.rs (hunk 2 above)
Comment (lemaitre-aneo) [reply packages/rust/armonik/src/client/sessions.rs:111]: The idea was to always use `session::Raw` on the user side. Here, we are in the session module, so it's fine.

**Comment by lemaitre-aneo** [comment]
> Globally, a lot of documentation has not been provided. Do we implement a policy for which rust doc should be present ?

I have not written any doc on the messages that do not have any doc in the protos.

> remember to setup `cargo fmt`

Will do

File: packages/rust/armonik/examples/client.rs
Comment (lemaitre-aneo) [reply packages/rust/armonik/examples/client.rs:1]: sure, it's just that I needed a code to test at some point.

**Review by aneojgurhem — APPROVED** [review]

**Pull request merged**